src/hotspot/share/compiler/compilerDefinitions.cpp

```diff
@@ -123,6 +123,16 @@ intx CompilerConfig::scaled_freq_log(intx freq_log) {
   return scaled_freq_log(freq_log, CompileThresholdScaling);
 }
 
+// For XXXThreshold flags, which all have a valid range of [0 .. max_jint]
+intx CompilerConfig::jvmflag_scaled_compile_threshold(intx threshold) {
+  return MAX2((intx)0, MIN2(scaled_compile_threshold(threshold), (intx)max_jint));
+}
+
+// For XXXNotifyFreqLog flags, which all have a valid range of [0 .. 30]
+intx CompilerConfig::jvmflag_scaled_freq_log(intx freq_log) {
+  return MAX2((intx)0, MIN2(scaled_freq_log(freq_log), (intx)30));
+}
```

Comment:
If the passed in arguments are out of range this appears to cap them within range but the initial error is never made known to the caller. ??

Reply:
This code is called only by `CompilerConfig::set_compilation_policy_flags()` to scale 15 flags according to `CompileThresholdScaling`. If the user sets `CompileThresholdScaling` to be too high, then some of the calculated values will be capped.

@veresov @TobiHartmann - do you think some warning messages are needed here?

My thinking is that `CompileThresholdScaling` is a somewhat magical knob. If the user really wants to know what's happening, they should add `-XX:+PrintFlagsFinal` and see the effect on all the affected flags.

Reply:
Okay - whether this should produce warnings is a distinct issue for compiler folk to decide.

Reply:
I think it's fine to cap the values of the individual thresholds at their maximum without a warning, when `CompileThresholdScaling` is used.

Similar issues were already discussed in https://github.com/openjdk/jdk/pull/8501.

> I suggest to address this issue in a separate RFE.

@tobiasholenstein did you file that separate RFE?

Reply:
> I think it's fine to cap the values of the individual thresholds at their maximum without a warning, when `CompileThresholdScaling` is used.
> 
> Similar issues were already discussed in #8501.
> 
> > I suggest to address this issue in a separate RFE.
> 
> @tobiasholenstein did you file that separate RFE?

No I did not file an RFE back then. But it seems like this PR resolves the issue

Reply:
Even better, thanks for verifying!